src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -0,0 +1,492 @@
+                    }
+                }
+
+                private void AnalyzeFieldOrPropertyReference(OperationAnalysisContext operationAnalysisContextContext)
+                {
+                    var fieldOrPropertyReference = operationAnalysisContextContext.Operation;
+                    if (!_delegateAssignedToFieldOrProperty &&
+                        fieldOrPropertyReference.Type.IsDelegateType() &&
+                        fieldOrPropertyReference.Parent is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target == fieldOrPropertyReference)
+                    {
+                        _delegateAssignedToFieldOrProperty = true;
+                    }
+                }
+
+                /// <summary>
+                /// Method invoked in <see cref="AnalyzeOperationBlockEnd(OperationBlockAnalysisContext)"/>
+                /// for each operation block to determine if we should analyze the operation block or bail out.
+                /// </summary>
+                private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol)
+                {
+                    switch (operationBlock.Kind)
+                    {
+                        case OperationKind.None:
+                        case OperationKind.ParameterInitializer:
+                            // Skip blocks from attributes (which have OperationKind.None) and parameter initializers.
+                            // We don't have any unused values in such operation blocks.
+                            return false;
+                    }
+
+                    // We currently do not support points-to analysis, so we cannot accurately 
+                    // track delegate invocations for all cases.
+                    // We attempt to do our best effort delegate invocation analysis as follows:
+
+                    //  1. If we have no delegate creations or lambdas, our current analysis works fine,
+                    //     return true.
+                    if (!_hasDelegateCreationOrAnonymousFunction)
+                    {
+                        return true;
+                    }
+
+                    //  2. Bail out if we have a delegate escape via an assigment to a field/property reference.
+                    if (_delegateAssignedToFieldOrProperty)
+                    {
+                        return false;
+                    }
+
+                    //  3. Bail out if we have a conversion from a delegate type to a non-delegate type.
+                    //     We can analyze this correctly when we do points-to-analysis.
+                    if (_hasConversionFromDelegateTypeToNonDelegateType)
+                    {
+                        return false;
+                    }
+
+                    //  4. Bail out for method returning delegates or ref/out parameters of delegate type.
+                    //     We can analyze this correctly when we do points-to-analysis.
```

Comment:
point-to is mentioned many times.  but, afaik, it's not a well known term.  can you explain it in this PR?